django/db/backends/postgresql/base.py

```diff
@@ -321,11 +321,26 @@ def init_connection_state(self):
     @async_unsafe
     def create_cursor(self, name=None):
         if name:
-            # In autocommit mode, the cursor will be used outside of a
-            # transaction, hence use a holdable cursor.
-            cursor = self.connection.cursor(
-                name, scrollable=False, withhold=self.connection.autocommit
-            )
+            if is_psycopg3 and (
+                not (conn_params := self.settings_dict.get("OPTIONS"))
+                or conn_params.get("server_side_binding") is not True
```

Comment:
I get what you are trying to achieve here but it is a little bit hard to read. Maybe move the retrieval of options out of the if ala: `options = self.settings_dict.get("OPTIONS") or {}` (this should handle both the case where `OPTIONS` is set to `None` as opposed to a `.get("OPTIONS", {})`) and then you can simply use `.get` on that in the if statement.

No strong feeling though.

Reply:
I could see us go either way so lets defer to mergers. I say lets focus on the cursor class part since this is really just a workaround for the lack of `server_cursor_factory`.